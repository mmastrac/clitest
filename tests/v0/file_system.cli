#!/usr/bin/env clitest --v0

# Functional tests for project linking
# Key:
#   - # <comment>
#   - $ <command>
#   - ? <grok pattern>
#   - ! <literal>
#   - repeat { ... }
#   - option { ... }
#   - ignore { ... }
#   - choice { ... }
#   - unordered { ... }

#pattern FILENAME [\w\-\.]+

$ rm -rf /tmp/testing-dir || echo "no dir to remove"
$ mkdir /tmp/testing-dir

$ cd /tmp/testing-dir && ls -al
! total %{NUMBER}
! %{GREEDYDATA} .
! %{GREEDYDATA} ..

$ echo "hello world" > /tmp/testing-dir/file1
$ echo "world" > /tmp/testing-dir/file2

$ ls -al /tmp/testing-dir
!!!
total %{NUMBER}
%{GREEDYDATA} .
%{GREEDYDATA} ..
%{GREEDYDATA} file1
%{GREEDYDATA} file2
!!!

# Test with a pattern that matches a filename
$ ls -al /tmp/testing-dir
! total %{NUMBER}
repeat {
    ! %{GREEDYDATA} %{FILENAME}
}

# Test with a pattern that matches a filename
$ ls -al /tmp/testing-dir 1>&2
! total %{NUMBER}
repeat {
    ! %{GREEDYDATA} %{FILENAME}
}
