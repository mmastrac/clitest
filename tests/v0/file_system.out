Running <root>/file_system.cli ...
using tempdir: <tmp>

mkdir "testing-dir"
---
(no output)
---
✅ OK

ls -a1
---
.
..
testing-dir
---
✅ OK

printf "hello world\n" > testing-dir/file1
---
(no output)
---
✅ OK

printf "world\n" > testing-dir/file2
---
(no output)
---
✅ OK

ls -a1 testing-dir
---
.
..
file1
file2
---
✅ OK

ls -1 "testing-dir/"*
---
testing-dir/file1
testing-dir/file2
---
✅ OK

ls -a1 testing-dir
---
.
..
file1
file2
---
✅ OK

ls -a1 testing-dir 1>&2
---
.
..
file1
file2
---
✅ OK

(cleanup) removing <tmp> && cd <root>

<root>/file_system.cli PASSED
